### ISSUE TITLE
RenkoBar Consolidator not functioning as expected

### ISSUE THREAD

**Issue by simonsonjack** [QuantConnect/Lean#2853]
#### Expected Behavior
Consolidator will build bars at appropriate intervals according to the argument passed to RenkoConsolidator(10) -- in this case, $10.

#### Actual Behavior
Despite a number of occasions where there is a $10 difference, the consolidator fails to build bars until late in the time series -- that is, there are instances where the stock price fluctuates around $121 when it's starting price was around $111.

#### Potential Solution
Examine and update the behavior of the Renko Consolidator.

#### Reproducing the Problem
This code will reproduce the problem and show a RenkoBar plot overlaid with an asset price plot.
```
def Initialize(self):

        self.SetStartDate(2012,1, 1)  #Set Start Date
        self.SetEndDate(2013,1,1)    #Set End Date
        self.SetCash(100000)           #Set Strategy Cash

        self.AddEquity("SPY", Resolution.Hour)
        renko_bar = RenkoConsolidator(10)
        renko_bar.DataConsolidated += self.OnDataConsolidated
        self.SubscriptionManager.AddConsolidator('SPY', renko_bar)
        
        renko_plot = Chart('Renko')
        renko_plot.AddSeries(Series('Renko Bar', SeriesType.Bar, 4))
        renko_plot.AddSeries(Series('Stock', SeriesType.Bar, 0))        
    
    def OnData(self, data):
        
        if data.Dividends.ContainsKey('SPY'):
            return
        self.Plot('Renko', 'Stock', data['SPY'].Close)
        self.Log(str(self.Time) + '   SPY Ceiling: $' + str(np.ceil(data['SPY'].Close)))
        self.Log(str(self.Time) + '   SPY Floor: $' + str(np.floor(data['SPY'].Close)))
            
    def OnDataConsolidated(self, sender ,bar):
        self.Plot('Renko', 'Renko Bar', bar.Close)
        self.Log(str(self.Time) + '   Renko Bar: $' + str(bar.Close))
```

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

**Comment by adriantorrie** [comment]
@simonsonjack Is this still an issue, or has it been inherently fixed as part of #3764?

**Comment by amijinxed** [comment]
Still a problem.
